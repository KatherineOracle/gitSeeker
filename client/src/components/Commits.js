/* Display top 5 commits per repository 

Displays "No Commits." if there are no commits in the object.
*/

const Commits = ({ commits, dateSettings }) => {


  if (commits.length === 0) {
    return <p>No Commits.</p>;
  }

  return (
      <ul>
        {commits.map((commit, idxx) => {
          return (
            <li key={"commit-" + idxx}>
              <strong>{commit.message}</strong> <br />
              <small>
                {new Date(commit.date).toLocaleDateString(
                  "en-US",
                  dateSettings
                )}
              </small>
            </li>
          );
        })}
      </ul>
  );
};

export default Commits;
