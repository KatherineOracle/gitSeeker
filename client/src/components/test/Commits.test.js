import React from "react";
import Commits from '../Commits'
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer
    .create(<Commits commits={[{"message":"Make URLs relative","committer":"KatherineOracle","date":"2022-02-19T14:31:52Z"}]} dateSettings={{weekday: 'long', year: 'numeric', month: 'long', day: 'numeric'}} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
    });
