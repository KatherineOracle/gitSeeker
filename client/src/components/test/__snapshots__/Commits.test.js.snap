// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul>
  <li>
    <strong>
      Make URLs relative
    </strong>
     
    <br />
    <small>
      Saturday, February 19, 2022
    </small>
  </li>
</ul>
`;
